### PR TITLE
Radar-derived microphysics temperature tendencies similar to operational HRRR

### PR DESCRIPTION
**Pull request by SamuelTrahanNOAA**
## Description

This implements a feature of the operational HRRR, radar-derived microphysics temperature tendencies applied in the first N minutes of the forecast to improve clouds in the first few hours. This has been tested for several weeks of forecasts and found to improve clouds. Collaborators on the DA and workflow development of this are @AmandaBack-NOAA @Ruifang-Li @hu5970

Also in here is convection suppression similar to the operational RAP. That is experimental, as it has only undergone technical testing and one case study. The code will warn you if you try to use it.

The ufs-weather-model PR is at https://github.com/ufs-community/ufs-weather-model/pull/986

### Issue(s) addressed

https://github.com/ufs-community/ufs-weather-model/issues/985

## Testing

Regression tests passed on Jet with Intel, Hera with Intel, and Hera with GNU.

Feature was shown to improve results in several weeks of forecasts in the HRRR-like RRFS.

There are two new regression tests at the ufs-weather-model level to ensure this functionality does not break.

## Dependencies

- waiting on https://github.com/NCAR/ccpp-physics/pull/823

### CODE REVIEW

File: ccpp/data/GFS_typedefs.meta [hunk 1]
```diff
@@ -3381,6 +3381,12 @@
   dimensions = ()
   type = real
   kind = kind_phys
+[num_dfi_radar]
+  standard_name = num_dfi_radar
```
Comment (grantfirl) [inline ccpp/data/GFS_typedefs.meta]: Some of these new standard names can probably be improved and/or made to be consistent with https://github.com/ESCOMP/CCPPStandardNames/blob/main/StandardNamesRules.rst. For example, what does DFI stand for?

Comment (SamuelTrahanNOAA) [reply ccpp/data/GFS_typedefs.meta]: The names match those in WRF. I can make them more elaborate in the meta files if that is necessary.

File: ccpp/data/GFS_typedefs.meta [hunk 2]
```diff
@@ -3459,6 +3465,26 @@
   units = flag
   dimensions = ()
   type = logical
+[radar_tten_limits]
+  standard_name = allowed_bounds_of_radar_prescribed_tendencies
+  long_name = allowed bounds of prescribed microphysics temperature tendencies
+  units = K
```
Comment (grantfirl) [inline ccpp/data/GFS_typedefs.meta]: should this be K s-1?

Comment (SamuelTrahanNOAA) [reply ccpp/data/GFS_typedefs.meta]: Yes.

File: ccpp/data/GFS_typedefs.meta (hunk 1 above)
Comment (SamuelTrahanNOAA) [reply ccpp/data/GFS_typedefs.meta]: The variables now have more satisfying names.

File: ccpp/data/GFS_typedefs.F90 [hunk 3]
```diff
@@ -3737,6 +3761,62 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
     Model%flag_for_scnv_generic_tend = .true.
     Model%flag_for_dcnv_generic_tend = .true.
 
+    Model%fh_dfi_radar     = fh_dfi_radar
```
Comment (climbfuji) [inline ccpp/data/GFS_typedefs.F90:3765]: I wonder if that whole block 3764-3819 should go into its own subroutine.

Comment (SamuelTrahanNOAA) [reply ccpp/data/GFS_typedefs.F90:3765]: done

**Review by climbfuji — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**